Website/src/rendering/src/components/Products/ProductDetailPage.tsx

```diff
@@ -0,0 +1,340 @@
+import { useRouter } from 'next/router';
+import { Spec, Variant } from 'ordercloud-javascript-sdk';
+import { FormEvent, useCallback, useEffect, useState } from 'react';
+import useOcProductDetail from 'src/hooks/useOcProductDetail';
+import { createLineItem } from 'src/redux/ocCurrentCart';
+import { useAppDispatch, useAppSelector } from 'src/redux/store';
+import { logViewEvent } from 'src/services/CdpService';
+import ProductQuantityInput from './ProductQuantityInput';
+import ProductSpecField from './ProductSpecField';
+import { Product } from './Shop';
+
+type ProductProps = {
+  name: string;
+  price: number;
+  image_url: string;
+  brand: string;
+  sku: string;
+};
+
+type Widget = {
```

Comment:
If this is not specific to the product details component, could it please be moved to `DiscoverService`?